siriuspy/siriuspy/opticscorr/utils.py

```diff
@@ -1,43 +1,47 @@
 """Optics correction utilities module."""
 
 
-def get_config_name(acc, opticsparam):
-    """Get the configuration name in file."""
-    if not (acc in ['bo', 'si']):
-        raise ValueError("'acc' must be 'bo' or 'si' instance.")
-    if not (opticsparam in ['tune', 'chrom']):
-        raise ValueError("'opticsparam' must be 'tune' or 'chrom' instance.")
-
-    try:
-        f = open('/home/sirius/iocs-log/'+acc+'-ap-'+opticsparam+'corr/' +
-                 acc+'-'+opticsparam+'corr.txt', 'r')
-        config_name = f.read().strip('\n')
-        f.close()
-    except Exception:
-        f = open('/home/sirius/iocs-log/'+acc+'-ap-'+opticsparam+'corr/' +
-                 acc+'-'+opticsparam+'corr.txt', 'w+')
-        config_name = get_default_config_name(acc, opticsparam)
-        f.write(config_name)
-        f.close()
-    return config_name
-
-
-def set_config_name(acc, opticsparam, config_name):
-    """Set the configuration name in file."""
-    if not (acc in ['bo', 'si']):
-        raise ValueError("'acc' must be 'bo' or 'si' instance.")
-    if not (opticsparam in ['tune', 'chrom']):
-        raise ValueError("'opticsparam' must be 'tune' or 'chrom' instance.")
-    if not isinstance(config_name, str):
-        raise TypeError("'config_name' must be a string.")
-
-    f = open('/home/sirius/iocs-log/'+acc+'-ap-'+opticsparam+'corr/' +
-             acc+'-'+opticsparam+'corr.txt', 'w+')
-    f.write(config_name)
-    f.close()
-
-
-def get_default_config_name(self, acc, opticsparam):
+class HandleConfigNameFile:
+    """Class to handle config_name file."""
+
+    def __init__(self, acc, opticsparam):
+        """Init."""
+        if not (acc in ['BO', 'SI']):
+            raise ValueError("'acc' must be 'BO' or 'SI' instance.")
+        if not (opticsparam in ['tune', 'chrom']):
+            raise ValueError(
+                "'opticsparam' must be 'tune' or 'chrom' instance.")
+        self.acc = acc.lower()
+        self.opticsparam = opticsparam
+        self.fpath = '/home/sirius/iocs-log/' + self.acc + '-ap-' + \
+            opticsparam+'corr/'
+        self.fname = self.fpath + self.acc + '-' + opticsparam + 'corr.txt'
+
+    def get_config_name(self):
+        """Get the configuration name in file."""
+        try:
+            fil = open(self.fname, 'r')
```

Comment:
try to use context manager here. Instead of 

```
fil = open(self.fname, 'r')
config_name = fil.read().strip('\n')
fil.close()

```
you could simply do:
```
with  open(self.fname, 'r') as fil:
    config_name = fil.read().strip('\n')
```

Reply:
done!